functions/src/delete_shared_schedule.ts

```diff
@@ -61,38 +71,38 @@ export const deleteInvitationFromFriend = functions.https.onRequest(
         }
 
         // Get user UID from the decoded token
-        const friendId = decodedToken.uid;
+        const requesterId = decodedToken.uid;
 
-        const senderRes = await schedulesCollection.doc(senderId).get();
-        const senderData: Version3ScheduleData | undefined =
-          senderRes.data() as Version3ScheduleData | undefined;
+        const senderId = owner ? requesterId : otherUserId;
+        const friendId = owner ? otherUserId : requesterId;
 
```

Comment:
validate that the sender is actually the user that we received the token of. Also validate that the receiver is actually the user we received the token of if owner is set to false.